### PR TITLE
Optimize MetisTuner

### PR DESCRIPTION
**Pull request by leelaylay**
1. Fix data type bug.
2. Add parallel support.
3. Increase default `selection_num_starting_points` value.
4. Add exploration_probability for selecting parameter from next_candidate.
5. Pretty the code.

### CODE REVIEW

File: src/sdk/pynni/nni/metis_tuner/metis_tuner.py [hunk 1]
```diff
@@ -368,21 +380,23 @@ def _selection(self, samples_x, samples_y_aggregation, samples_y,
...
-    
+
                         if next_improvement > temp_improvement:
-                            # logger.info("DEBUG: \"next_candidate\" changed: \
-                            #                 lowest mu might reduce from %f (%s) to %f (%s), %s\n" %\
-                            #                 lm_current['expected_mu'], str(lm_current['hyperparameter']),\
-                            #                 threads_result['expected_lowest_mu'],\
-                            #                 str(threads_result['candidate']['hyperparameter']),\
-                            #                 threads_result['candidate']['reason'])
+                            logger.infor("DEBUG: \"next_candidate\" changed: \
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/metis_tuner/metis_tuner.py]: there is a bug here

File: src/sdk/pynni/nni/metis_tuner/metis_tuner.py [hunk 2]
```diff
@@ -392,6 +406,9 @@ def _selection(self, samples_x, samples_y_aggregation, samples_y,
                                   'expected_mu': expected_mu, 'expected_sigma': expected_sigma}
 
         outputs = self._pack_output(lm_current['hyperparameter'])
+        if outputs in self.history_parameters:
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/metis_tuner/metis_tuner.py]: could you add some logic about "take next config as exploration step randomly."

File: src/sdk/pynni/nni/metis_tuner/metis_tuner.py (hunk 1 above)
Comment (leelaylay) [reply src/sdk/pynni/nni/metis_tuner/metis_tuner.py]: Thank you.

File: src/sdk/pynni/nni/metis_tuner/metis_tuner.py (hunk 2 above)
Comment (leelaylay) [reply src/sdk/pynni/nni/metis_tuner/metis_tuner.py]: OK~

File: src/sdk/pynni/nni/metis_tuner/metis_tuner.py [hunk 3]
```diff
@@ -245,7 +250,7 @@ def receive_trial_result(self, parameter_id, parameters, value):
 
             # calculate y aggregation
             median = get_median(temp_y)
-            self.samples_y_aggregation[idx] = median
+            self.samples_y_aggregation[idx] = [median]
```
Comment (leckie-chn) [inline src/sdk/pynni/nni/metis_tuner/metis_tuner.py]: so this solves the ValueError of "setting an array element with a sequence"?

File: src/sdk/pynni/nni/metis_tuner/metis_tuner.py [hunk 4]
```diff
@@ -67,40 +65,45 @@ class MetisTuner(Tuner):
     """
 
     def __init__(self, optimize_mode="maximize", no_resampling=True, no_candidates=True,
-                 selection_num_starting_points=10, cold_start_num=10):
+                 selection_num_starting_points=600, cold_start_num=10, exploration_probability=0.1):
```
Comment (leckie-chn) [inline src/sdk/pynni/nni/metis_tuner/metis_tuner.py]: why adjust those default parameters? any changes introduced regarding the tuner's default behaviour?

File: src/sdk/pynni/nni/metis_tuner/metis_tuner.py (hunk 3 above)
Comment (xuehui1991) [reply src/sdk/pynni/nni/metis_tuner/metis_tuner.py]: yes

File: src/sdk/pynni/nni/metis_tuner/metis_tuner.py (hunk 4 above)
Comment (xuehui1991) [reply src/sdk/pynni/nni/metis_tuner/metis_tuner.py]: we adjust this parameter due to the experiment result.

**Pull request merged**